File: Stack/150_ReversePolishNotation/main.py
def evalRPN(tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        if not tokens: return 0
        stack = []
        for item in tokens:
            if item in "+-*/":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(str(int(eval(val2+item+val1))))
            else:
                stack.append(item)
            
        return int(stack.pop())

tokens = ["4","13","5","/","+"]
print(evalRPN(tokens))

File: Stack/150_ReversePolishNotation/main.cpp
class Solution {
public:
    
    int evalRPN(vector<string>& tokens) {
        stack<int> stack;
        static unordered_set<string> ops({"+", "-", "*", "/"});
        for (auto t : tokens) {
            if (ops.find(t) == ops.end()) stack.push(stoi(t));
            else {
                int b = stack.top(); stack.pop();
                int a = stack.top(); stack.pop();
                if (t == "+") stack.push(a + b);
                else if (t == "-") stack.push(a - b);
                else if (t == "*") stack.push(a * b);
                else stack.push(a / b);
            }
        }
        return stack.top();
    }
};